### PR TITLE
fix AutoLoginToken serialization

### PR DESCRIPTION
**Pull request by ezprit**
This PR fixes AutoLoginToken serialization for symfony/security-core > 4.3

https://github.com/symfony/security-core/commit/f948fedde6c1b86d9a8b16d8016ce3376be20f74

### CODE REVIEW

File: src/Jmikola/AutoLogin/Authentication/Token/AutoLoginToken.php [hunk 1]
```diff
@@ -54,24 +54,15 @@ public function getCredentials() : string
...
-    /**
-     * @see Symfony\Component\Security\Core\Authentication\Token\AbstractToken::unserialize()
-     */
-    public function unserialize($str)
+    public function __unserialize(array $data): void
     {
-        [$this->key, $this->providerKey, $parentStr] = unserialize($str);
-        parent::unserialize($parentStr);
+        [$this->key, $this->providerKey, $parentData] = $data;
+        $parentData = \is_array($parentData) ? $parentData : unserialize($parentData);
```
Comment (jmikola) [inline src/Jmikola/AutoLogin/Authentication/Token/AutoLoginToken.php:65]: Quoting https://github.com/symfony/security-core/commit/f948fedde6c1b86d9a8b16d8016ce3376be20f74#diff-c5e43b78aa0cd524bf4965e55c5839b0e4eebc0afde1fd0ba45deb946d1734d9R187:

> There is no need to unserialize any entry in $data, they are already ready-to-use.

In that case, would you agree that this line is redundant and should be removed?

File: src/Jmikola/AutoLogin/Authentication/Token/AutoLoginToken.php [hunk 2]
```diff
@@ -54,24 +54,15 @@ public function getCredentials() : string
...
-     */
-    public function serialize()
+    public function __serialize(): array
     {
-        return serialize(array(
-            $this->key,
-            $this->providerKey,
-            parent::serialize(),
-        ));
+        return [$this->key, $this->providerKey, parent::__serialize()];
```
Comment (jmikola) [inline src/Jmikola/AutoLogin/Authentication/Token/AutoLoginToken.php:59]: This matches the example provided in https://github.com/symfony/security-core/commit/f948fedde6c1b86d9a8b16d8016ce3376be20f74#diff-c5e43b78aa0cd524bf4965e55c5839b0e4eebc0afde1fd0ba45deb946d1734d9R145, so I trust it's correct.

**Review by jmikola — CHANGES_REQUESTED** [review]
Sorry for the delayed review. I have one question on a possible change before merging.

**Comment by atiernan** [comment]
I'm also experiencing this issue. Is there any chance you could check over this PR @jmikola?

**Comment by jmikola** [comment]
Merged and released as [2.0.1](https://github.com/jmikola/AutoLogin/releases/tag/2.0.1).

**Pull request merged**